internal/cluster/cluster.go

```diff
@@ -811,20 +834,14 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 		models.ClusterStatusInsufficient,
 		models.ClusterStatusReady,
 	}
-	var cluster common.Cluster
-	if err := db.Select("id, status").Take(&cluster, "id = ?", clusterID.String()).Error; err != nil {
-		return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Getting cluster %s", clusterID.String()))
-	}
-
 	if !funk.ContainsString(allowedStates, swag.StringValue(cluster.Status)) {
 		return nil
 	}
 
-	var hosts []*models.Host
-	if err := db.Order("id").Select("id, connectivity, inventory").Find(&hosts, "cluster_id = ? and status <> ?", clusterID.String(), models.HostStatusDisabled).Error; err != nil {
-		return common.NewApiError(http.StatusInternalServerError, errors.Wrapf(err, "Getting hosts for cluster %s", clusterID.String()))
-	}
-
+	hosts := cluster.Hosts
+	sort.Slice(hosts, func(i, j int) bool {
```

Comment:
it would help to have a comment explaining why sorting is needed.

Reply:
OK.  Will change in a later PR